### PR TITLE
Prevents private functions call from outside a ezjscServerFunctions class

### PR DESCRIPTION
**Pull request by alafon**
Considering that we have a class AServerFunctions extending ezjscServerFunctions

``` php

<?php

/**
* A class to call server side functions using ajax through eZJscore
*
*/
class AServerFunctions extends ezjscServerFunctions
{
    public static function callableFunctionA( $args )
    {
        return self::notACallableFunction();
    }

    public static function callableFunctionB( $args )
    {
        return self::notACallableFunction();
    }

    private static function notACallableFunction( )
    {
        return "something";
    }
}

```

Functions `callableFunctionA` and `callableFunctionB` are public static & callable using ezjscore/call/aserver::callableFunctionA

but `notACallableFunction` should not be callable by ezjscore/call since it's a private function. 

ezjscServerRouter::staticHasFunction() should not only check if the function exists, this commit aims to fix this

### CODE REVIEW

File: classes/ezjscserverrouter.php [hunk 1]
```diff
@@ -197,7 +197,7 @@ public static function staticHasFunction( $className, $functionName, $isTemplate
         }
         else
         {
-            return method_exists( $className, $functionName );
+            return is_callable( "{$className}::{$functionName}" );
```
Comment (lolautruche) [inline classes/ezjscserverrouter.php]: Are you sure this works with PHP 5.2.x ?

Comment (alafon) [inline classes/ezjscserverrouter.php]: Looks like : [php.net](http://fr2.php.net/manual/en/function.is-callable.php)
Actually, I did not make a lot of investigation, I just check on which version this function is available and it seems that it's ok. There also might be another solution to get the method's visibility value but I did not find it.

**Comment by andrerom** [comment]
```
// Type 4: Static class method call (As of PHP 5.2.3)
call_user_func('MyClass::myCallbackMethod');
```

ref :http://php.net/manual/en/language.types.callable.php
So it needs to be

```
return is_callable( array( $className, $functionName ) );
```

**Comment by alafon** [comment]
Fixed

**Comment by lolautruche** [comment]
Thanks Arnaud !

**Comment by andrerom** [comment]
Thanks indeed :)

**Comment by alafon** [comment]
My pleasure !

**Pull request merged**